apps/web/src/core/providers/agent-manager.ts

```diff
@@ -0,0 +1,157 @@
+import { createProvider } from "puro"
```

Comment:
I think it would be helpful to organize all the unique dependencies of the agent demo under one folder (and same for the vm demo). do you think we could do that? (rather than mixing up all demo dependencies together in these general folders like `providers`)

Reply:
Let's do that - I was mirroring the structure in the extensions. I do prefer to have feature-dedicated/oriented directories, helps with isolation and refactoring